pkg/arvo/app/language-server.hoon

```diff
@@ -93,107 +109,71 @@
 ::
 |_  bow=bowl:gall
 ::
-++  parser
-  =,  dejs:format
-  |^
-  %:  ot
-    uri+so
-    :-  %data
-    %-  of
-    :~  sync+sync
-        completion+position
-        commit+ni
-        hover+position
-    ==
-    ~
-  ==
-  ::
-  ++  sync
-    %-  ar
-    %:  ou
-      range+(uf ~ (pe ~ range))
-      'rangeLength'^(uf ~ (pe ~ ni))
-      text+(un so)
-      ~
-    ==
-  ::
-  ++  range
-    %:  ot
-      start+position
-      end+position
-      ~
-    ==
-  ::
-  ++  position
-    %:  ot
-      line+ni
-      character+ni
-      ~
-    ==
-  --
-::
 ++  json-response
   |=  [eyre-id=@ta jon=json]
   ^-  (list card)
   (give-simple-payload:app eyre-id (json-response:gen (json-to-octs jon)))
 ::
-::  +handle-http-request: received on a new connection established
+++  give-rpc-notification
+  |=  res=out:notification:lsp-sur
+  ^-  (list card)
+  :_  ~
+  [%give %fact ~[/primary] %language-server-rpc-notification !>(res)]
 ::
-++  handle-http-request
-  |=  [eyre-id=@ta =inbound-request:eyre]
+++  on-notification
+  |=  not=all:notification:lsp-sur
   ^-  (quip card _state)
-  ?>  ?=(^ body.request.inbound-request)
-  =/  =lsp-req
-    %-  parser
-    (need (de-json:html q.u.body.request.inbound-request))
-  =/  buf  (~(gut by bufs) uri.lsp-req *wall)
-  =^  cards  buf
-    ?-  +<.lsp-req
-      %sync        (handle-sync buf eyre-id +>.lsp-req)
-      %completion  (handle-completion buf eyre-id +>.lsp-req)
-      %commit      (handle-commit buf eyre-id uri.lsp-req)
-      %hover       (handle-hover buf eyre-id +>.lsp-req)
+  =^  cards  state
+    ?+  -.not  [~ state]
+        %text-document--did-open  (handle-did-open +.not)
+        %text-document--did-change  (handle-did-change +.not)
+        %text-document--did-save  (handle-did-save +.not)
+        %text-document--did-close  (handle-did-close +.not)
```

Comment:
One last thing: we generally align the right side of these at furthest-right column (ie all at the level of `(handle-did-change +.not)`)